nemo/collections/speechlm/modules/asr_module.py

```diff
@@ -0,0 +1,135 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import Optional
+
+import torch
+import torch.nn as nn
+from megatron.core.model_parallel_config import ModelParallelConfig
+from megatron.core.transformer.module import MegatronModule
+from megatron.core.transformer.transformer_config import TransformerConfig
+from omegaconf import OmegaConf
+
+import nemo.collections.asr as nemo_asr
+from nemo.collections.speechlm.utils import to_dict_config
+from nemo.core.classes.common import Serialization
+from nemo.core.classes.module import NeuralModule
+from nemo.lightning import io
+from nemo.utils import logging, model_utils
+
+
+class MCoreASRModule(MegatronModule):
```

Comment:
Can you add some documentation what is the purpose of this class?

Reply:
done